workflow-core/src/main/java/com/squareup/workflow/WorkflowIdentifier.kt

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2020 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.workflow
+
+import okio.BufferedSink
+import okio.BufferedSource
+import okio.EOFException
+import kotlin.LazyThreadSafetyMode.PUBLICATION
+import kotlin.reflect.KClass
+
+/**
+ * Represents a [Workflow]'s "identity" and is used by the runtime to determine whether a workflow
+ * is the same as a workflow as was rendered in a previous render pass, in which case its state
+ * should be re-used, or if it's a new workflow and needs to be started.
+ *
+ * The identity consists primarily of the workflow's concrete type (i.e. the class that implements
+ * the [Workflow] interface). Two workflows of the same concrete type are considered identical.
+ *
+ * However, if a workflow class implements [ImpostorWorkflow], the identifier will also include
+ * that workflow's [ImpostorWorkflow.realIdentifier].
+ *
+ * @constructor
+ * The workflow's type must be passed _either_ as a [KClass] or a string, but only one must be
```

Comment:
Is this stale? The `type` param below isn't nullable.

Reply:
yep